ClientRuntimes/Ruby/ms-rest-azure/lib/ms_rest_azure/azure_application_credentials.rb

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+
+module MsRestAzure
+  #
+  # Class which represents an Azure token credentials.
+  #
+  class AzureApplicationCredentials < MsRest::ServiceClientCredentials
+
+    TOKEN_ACQUIRE_URL = 'https://login.windows.net/{tenant_id}/oauth2/token'
```

Comment:
The login authority should be settable, it is different for azure china, azure stack, government cloud